Makefile

```diff
@@ -48,133 +51,93 @@ help:
 	} \
 	{ lastLine = $$0 }' $(MAKEFILE_LIST)
 
-.PHONY: build-and-push-fuse-sidecar-img
-## build FUSE sidecar container used in Argo workflows
-build-and-push-fuse-sidecar-img:
-	@echo 'building fuse sidecar container'
-	docker build \
-		--progress plain \
-		-t gcr.io/onec-co/datamon-fuse-sidecar \
-		-t gcr.io/onec-co/datamon-fuse-sidecar:${GITHUB_USER}-$$(date '+%Y%m%d') \
-		-t gcr.io/onec-co/datamon-fuse-sidecar:$(subst /,_,$(GIT_BRANCH)) \
-		-t gcr.io/onec-co/datamon-fuse-sidecar:$(RELEASE_TAG) \
-		-t gcr.io/onec-co/datamon-fuse-sidecar:$(RELEASE_TAG_LATEST) \
-		--ssh default \
-		-f sidecar.Dockerfile \
-		.
-	docker push gcr.io/onec-co/datamon-fuse-sidecar
+.PHONY: build-target
+build-target:
+	@echo '$(GREEN)building $(YELLOW)$(BUILD_TARGET)$(GREEN) container...$(RESET)'
+	@echo "$(GREEN)with tags: $(TAGS)$(RESET)"
+	$(BUILDER) -f $(DOCKERFILE) $(BUILD_ARGS) -t $(BUILD_TARGET) \
+	$(shell for tag in $(TAGS) ; do echo "-t $(BUILD_TARGET):$$tag" ; done) \
+	.
+
+.PHONY: push-target
+push-target:
+	docker push $(BUILD_TARGET)
+
+.PHONY: build-datamon-binaries
+build-datamon-binaries: BUILD_TARGET=datamon-binaries
+build-datamon-binaries:
+	@make build-target \
```

Comment:
had to look up [the manual](https://www.gnu.org/software/make/manual/html_node/Recursion.html) on this.

i am not a make wizard, and i do not want to be a make wizard (`mk` being a different matter).
anyway, all opinions aside, `make` (in all its flavors, except perhaps automake) is worthwhile to know something about, so i've some readings.

specific gleanings from the manual appear to be

* [`$(MAKE)`](https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable) instead of `@make`
* [export](https://www.gnu.org/software/make/manual/html_node/Variables_002fRecursion.html#Variables_002fRecursion) as in `export DOCKERFILE = binaries.Dockerfile` as its own statement preceding the recursive use via `$(MAKE)` rather than [overriding](https://www.gnu.org/software/make/manual/html_node/Overriding.html#Overriding) via shell-like `make` invocation

Reply:
@ransomw1c done for `$(MAKE)`.
Regarding export I more nuanced. I've tested more following your recommendation and this changed a lot the behavior. I've now added some `export` directives for almost all targets, but the one that is precisely cascading, for which I need an override.